outputs/hn_zones/comparison_of_hn_zones.py

```diff
@@ -0,0 +1,447 @@
+
+4. **Data Export**:
+   - Outputs GeoDataFrame with coverage data, LSOA lists, and processed suitability scores to GeoPackage, JSON, Parquet, and CSV formats.
+   - Logs detailed metrics and analysis steps.
+
+**Outputs**:
+- GeoPackage: 'liverpool_with_desnz_hn_lsoa.gpkg'
+- JSON: 'liverpool_hp_suitability_lsoas.json'
+- Parquet: 'liverpool_hp_suitability_scores_with_desnz'
+- Log: 'script_output.log'
+
+**How to Run the Script**:
+To run the script, use the following command:
+python comparison_of_hn_zones.py [--optional_threshold OPTIONAL_THRESHOLD]
+"""
+
+import geopandas as gpd
+import pyogrio
+import polars as pl
+from typing import Tuple, List, Callable, Optional
+import logging
+import json
+import os
+from shapely.geometry import Polygon
+from asf_heat_pump_suitability import PROJECT_DIR
+import argparse
+
+
+# Define file paths
+# LIVERPOOL_GPKG_PATH = "s3://asf-heat-pump-suitability/heat_network_desnz_data/heat-network-zone-map-Liverpool.gpkg"
+# LSOA_SHP_PATH = "s3://asf-heat-pump-suitability/source_data/Lower_layer_Super_Output_Areas_2021_EW_BFE_V9_-9107090204806789093/LSOA_2021_EW_BFE_V9.shp"
+# NESTA_HP_SUITABILITY_PARQUET_PATH = "s3://nesta-open-data/asf_heat_pump_suitability/2023Q4/20240925_2023_Q4_EPC_heat_pump_suitability_per_lsoa.parquet"
+LA_TO_ANALYSE = "Liverpool"
+LIVERPOOL_GPKG_PATH = "heat-network-zone-map-Liverpool.gpkg"
+LSOA_SHP_PATH = "LSOA_2021_EW_BFE_V9.shp"
+NESTA_HP_SUITABILITY_PARQUET_PATH = (
+    "20240925_2023_Q4_EPC_heat_pump_suitability_per_lsoa.parquet"
+)
+
+
+def setup_logging_and_file_path(
+    output_dir: str, log_filename: str = "script_output.log", level: int = logging.INFO
+) -> None:
+    """
+    Set up logging configuration to log messages to both a file and the console.
+
+    Args:
+        output_dir (str): Path to the directory where the log file should be saved.
+        log_filename (str): Name of the log file. Defaults to "script_output.log".
+        level (int): Logging level, e.g., logging.INFO or logging.DEBUG. Defaults to logging.INFO.
+    """
+    # Ensure the output directory exists
+    os.makedirs(output_dir, exist_ok=True)
+
+    # Define the full path to the log file
+    log_file_path = os.path.join(output_dir, log_filename)
+
+    # Clear existing logging handlers if any
+    logging.getLogger().handlers.clear()
+
+    # Set up logging configuration
+    logging.basicConfig(
+        level=level,
+        format="%(levelname)s: %(message)s",
+        handlers=[
+            logging.FileHandler(log_file_path, mode="w"),  # Overwrite log file each run
+            logging.StreamHandler(),
+        ],
+    )
+    logging.info(f"Logging setup complete. Logs are saved to {log_file_path}.")
+
+
+def load_hn_geodata(
+    desnz_hn_gpkg_path: str, lsoa_shp_path: str
+) -> Tuple[gpd.GeoDataFrame, List[str]]:
+    """
+    Load the HN GeoPackage file and perform a spatial join with LSOA polygons to add a column of LSOA codes.
```

Comment:
```suggestion
    Load the Heat Networks GeoPackage file and perform a spatial join with LSOA polygons to add a column of LSOA codes.
```
We are also calculating the intersecting area so I think we should add that to the docstring :)

Reply:
good point!